src/Components/ArtsyShippingEstimate.tsx

```diff
@@ -170,6 +176,104 @@ export const ArtsyShippingEstimate = ({
   )
 }
 
+interface PriceEstimate {
+  destination: string
+  minPrice: number
+  maxPrice?: number | null
+  currency: string
+}
+
+interface UseWidgetObserverProps {
+  onViewEstimatedPrice: (price: PriceEstimate) => void
+}
+
+const useWidgetObserver = ({
+  onViewEstimatedPrice,
+}: UseWidgetObserverProps) => {
+  const widgetObserver = useRef<MutationObserver | null>(null)
+  const [visiblePrice, setVisiblePrice] = useState<PriceEstimate | null>(null)
+
+  const extractEstimateFromDom = useCallback((): PriceEstimate | null => {
+    const priceAmountEl = document.getElementsByClassName(
+      "artajs__modal__quotes__price__amount",
```

Comment:
This is so hacky! Relying on the dom selectors here. But it's a test - I'm ok with it :)

Reply:
yeah...